src/com/java/multithreading/TestBlockingQueue.java

```diff
@@ -0,0 +1,20 @@
+package com.java.multithreading;
+
+public class TestBlockingQueue {
+
+  public static void main(String[] args) {
+
+    BlockingQueue queue = new BlockingQueue(10);
+    Thread consumer1 = new Thread(new Consumer(queue));
+    Thread producer = new Thread(new Producer(queue));
+
+    Thread consumer2 = new Thread(new Consumer(queue));
+    Thread consumer3 = new Thread(new Consumer(queue));
+
+    producer.start();
+    consumer1.start();
+    consumer2.start();
+    consumer3.start();
```

Comment:
You should join your main thread to both producers and consumers also,  otherwise your main thread will end first, and depending upon the implementation of JVM, your application can behave weirdly.